### PR TITLE
Update and add several Intel extensions

### PR DESCRIPTION
**Pull request by MrSidims**
- Update SPV_INTEL_fpga_loop_controls to rev I
Spec: https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_fpga_loop_controls.asciidoc
It also fixes a bug for NoFusionINTEL, which shouldn't have a parameter

- Add SPV_INTEL_fpga_invocation_pipelining_attributes
Spec: https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_fpga_invocation_pipelining_attributes.asciidoc

- Add SPV_INTEL_fpga_dsp_control
Spec: https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_fpga_dsp_control.asciidoc

- Add SPV_INTEL_runtime_aligned
Spec:  https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_runtime_aligned.asciidoc

### CODE REVIEW

**Comment by MrSidims** [comment]
Associated PRs: [SPIRV-LLVM-Translator](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1657), SPIRV-Tools TBD

**Comment by MrSidims** [comment]
Tagging @bashbaug

File: include/spirv/unified1/spirv.core.grammar.json [hunk 1]
```diff
@@ -9086,6 +9086,23 @@
...
+          "version" : "None"
+        },
+        {
+          "enumerant" : "LoopCountINTEL",
+          "value" : "0x1000000",
+          "parameters" : [
+            { "kind" : "LiteralInteger" }
+          ],
+          "capabilities" : [ "FPGALoopControlsINTEL" ],
+          "extensions" : [ "SPV_INTEL_fpga_loop_controls" ],
```
Comment (bashbaug) [inline include/spirv/unified1/spirv.core.grammar.json:9100]: Note: This follows the same pattern used by the other loop controls added for this extension, but I think our current best practice is to gate the capability by the extension, then anything else by the capability only.  Is it sufficient to only include the capability here?

**Review by bashbaug — APPROVED** [review]
Mostly LGTM, matches the values described in the extension docs, just have one stylistic question regarding the loop control bits.

File: include/spirv/unified1/spirv.core.grammar.json (hunk 1 above)
Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json]: Indeed, remove extra addition of the capability for all SPV_INTEL_fpga_loop_controls extension entries

Comment (bashbaug) [reply include/spirv/unified1/spirv.core.grammar.json]: Thanks, your latest change removed the redundancy, but I think in the opposite direction it should have: I think it should have removed the extension and left the capability.

Comment (MrSidims) [reply include/spirv/unified1/spirv.core.grammar.json]: Oh I see, thanks!

Comment (bashbaug) [reply include/spirv/unified1/spirv.core.grammar.json]: Thanks, latest version looks great 👍

**Comment by bashbaug** [comment]
LGTM, I think this is ready to go after working group review.

**Pull request merged**